config/main.py

```diff
@@ -2323,6 +2323,140 @@ def reload(ctx, no_dynamic_buffer, dry_run, json_data):
+                raise click.Abort()
+            asic_id_suffix = str(asic_id)
+
+            config_db = ConfigDBConnector(use_unix_socket_path=True, namespace=ns)
+
+        config_db.connect()
+        if is_dynamic_buffer_enabled(config_db):
+            buffer_template_file = os.path.join(hwsku_path, asic_id_suffix, "buffers_dynamic.json.j2")
+        else:
+            buffer_template_file = os.path.join(hwsku_path, asic_id_suffix, "buffers.json.j2")
+
+        if not os.path.isfile(buffer_template_file):
+            click.secho("Buffer definition template not found at {}".format(buffer_template_file), fg="yellow")
+            ctx.abort()
+            raise click.Abort()
+
+        qos_template_file = os.path.join(hwsku_path, asic_id_suffix, "qos.json.j2")
+
+        if not os.path.isfile(qos_template_file):
+            click.secho("QoS definition template not found at {}".format(qos_template_file), fg="yellow")
+            raise click.Abort()
+
+        cmd_ns = "" if ns is DEFAULT_NAMESPACE else "-n {}".format(ns)
+        command = "{} {} {} -t {},config-db -t {},config-db -y {} --print-data".format(
+            SONIC_CFGGEN_PATH, cmd_ns, from_db, buffer_template_file, qos_template_file, sonic_version_file
+        )
+        jsonstr = clicommon.run_command(command, display_cmd=False, return_cmd=True)
+
+        jsondict = json.loads(jsonstr)
+        port_table = jsondict.get('PORT')
+        if port_table:
+            ports_to_update = set(port_table.keys()).intersection(portset)
+            if not ports_to_update:
+                continue
+        else:
+            continue
+
+        port_qos_items = {}
+        # Handle all tables
+        for table_name in tables_single_index:
+            table_items = jsondict.get(table_name)
+            if table_items:
+                result = {}
+                for key, data in table_items.items():
+                    port = key.split('|')[0]
+                    if not port in ports_to_update:
+                        continue
+                    # Push the result to config-db
+                    if not port_qos_items.get(table_name):
+                        port_qos_items[table_name] = {}
+                    port_qos_items[table_name][key] = data
+
+        # Handle CABLE_LENGTH
+        # This table needs to be specially handled because the port is not the index but the field name
+        # The idea is for all the items that are in the config_db or template
+        table_name = 'CABLE_LENGTH'
+        cable_length_table = jsondict.get(table_name)
+        if cable_length_table:
+            for key, item in cable_length_table.items():
+                cable_length_from_template = {}
+                for port in ports_to_update:
+                    cable_len = item.get(port)
+                    if cable_len:
+                        cable_length_from_template[port] = cable_len
+                cable_length_from_db = config_db.get_entry(table_name, key)
```

Comment:
suppose port list `Ethernet0,Ethernet4`, `cable_length_from_template` contains `{"AZURE": {"Ethernet0": "5m", "Ethenret4": "5m"}`
ports in the CONFIG_DB.PORT table: Ethernet0, Ethernet4, Ethernet8, Ethernet12, Ethernet16..., `cable_length_from_db` contains `{"AZURE": {"Ethernet8": "5m", "Ethenret12": "5m", "Ethenret16": "5m"}`
and the result will be `{"AZURE": {"Ethernet0": "5m", "Ethenret4": "5m", "Ethernet8": "5m", "Ethenret12": "5m", "Ethenret16": "5m"}`